backend/sprintone/src/main/java/com/project/omega/controller/ProductController.java

```diff
@@ -0,0 +1,93 @@
+package com.project.omega.controller;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.project.omega.bean.Product;
+import com.project.omega.bean.User;
+import com.project.omega.exceptions.NoRecordsFoundException;
+import com.project.omega.exceptions.ProductNotFoundException;
+import com.project.omega.exceptions.UserNotFoundException;
+import com.project.omega.service.ProductService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping(value = "/product")
```

Comment:
I'm also not understanding all these endpoints... Wouldn't it make sense to just have a `/product` (or `/products`)  and `/product/{id}` endpoint supporting `PUT`, `POST`, `GET` and `DELETE` methods for edit, new, get and delete respectively? The search parameters can be query parameters in the `GET` method.

Although this still works

Reply:
Valid point, I will consider how best to consolidate the endpoints and update the rest docs accordingly.